### DISCUSSION TITLE
Filter funktion

### DISCUSSION THREAD

**Discussion by Mersag** [neuefische/questions#84] [Web] [answered]
Hallo alle zusammen , 

ich möchte von meiner **Card Page** mit einem Link auf die **Effect(Wirkung) Page** und möchte das in Effect Page aus meiner **db** die Arrays angezeigt werden.

**Screenshot** 

<img width="1440" alt="Bildschirmfoto 2022-08-18 um 14 29 35" src="https://user-images.githubusercontent.com/107836739/185395092-32964d86-a059-4ccd-bdf2-b8608bcd39ba.png">

Ich habe jetzt einiges versucht und habe es nicht geschafft. Wer kann mir dabei helfen? Vielen Dank im voraus:-)

Ich konnte noch kein Commit bzgl. Effekt Page ausführen wegen der Fehlermeldung . 

**Ziel:** Dass auf der effect.js Page die Arrays aus der db.js gerendert wird.
(https://github.com/Mersag/capstone-projekt)

**Comment by F-Kirchhoff** [comment, marked as answer]
An und für sich ist dein Ansatz schon richtig, nur die Syntax ist nicht ganz korrekt. Deine `callback function` innerhalb von `filter` ist leider fehlerhaft. Du kannst bei den Parametern deiner Funktion nicht die dot notation verwenden. Ausserdem musst du bei dem Parameter nicht auf das Array `db` zugreifen, das macht `filter` für dich!
```js
db
 .filter( dbEntry => dbEntry.effects.includes(....)) // Has to be changed to fit your db data structure
 .map( filteredEffect => <li> ... )
```

**Reply by Mersag** [reply]
Thank you for the answer. I will try this Solution ..:-)

**Reply by Mersag** [reply]
Good morning, :-) I try this path and I have also a syntax Problem ... Hmmmm, let me think about ...

**Reply by F-Kirchhoff** [reply]
übrigens, falls du schauen möchtest ob dein Effekt einer von mehreren Optionen ist (`körperlich` oder `energetisch` oder `geistig` etc) musst du die `includes` methode anders verwenden.
Im Moment schaust du ob in dem Effekt genau der string `"körperlich, energetisch, geistig,..."` enthalten ist, was er sehr warscheinlich nicht ist (sondern nur einzelne Adjektive?)
Falls dein Effekt nur ein Adjektiv ist musst du es wie folgt schreiben:
```js
["körperlich","energetisch","geistig","muskulär"].includes(effect) // if effect is one of the adjectives, it returns true
```

**Comment by alexclaes** [comment]
Hallo Meral,

wenn ich es richtig verstehe, willst du eine Liste von allen Effekten aus der DB anzeigen. 

Dann benötigst du kein `.filter` und kein `.includes`. Wahrscheinlich denkst du gerade zu kompliziert.

Die Variable `db` beinhaltet ein Array. Du kann einfach alle Elemente aus dem Array mit `.map` darstellen. Bitte vergesse auch nicht, das du bei `.map` immer die Property `key={}` angeben musst

**Reply by Mersag** [reply]
Das bedeutet, ich entferne meine Filterfunktion und .map (e) mit der Property {key}, was ich als User auf der Effekt Page sehen möchte.

**Reply by Mersag** [reply]
Danke auch dir, Alex, für deine Hilfe.
